src/Controller/FullCalendarController.php

```diff
@@ -0,0 +1,413 @@
+    $query = $this->database->select('node_field_data', 'n');
+    $query->fields('n', ['nid', 'title']);
+    $query->condition('n.status', NodeInterface::PUBLISHED);
+    $query->condition('n.type', 'branch');
+    $query->orderBy('n.title');
+    $query->addTag('openy_home_branch_get_locations');
+    $query->addTag('node_access');
+
+    return $query->execute()->fetchAllKeyed();
+  }
+
+  /**
+   * Provides a JSON response containing a list of branches.
+   *
+   * This method leverages the `getBranches` method to retrieve an associative
+   * array of branch IDs and titles, and then encapsulates this array within a
+   * JSON response.
+   *
+   * @return JsonResponse
+   *   A JsonResponse object that contains the branches in an associative array
+   *   format, where each branch ID is a key, and its corresponding title is
+   *   the value.
+   */
+  public function getBranchesOptions(): JsonResponse {
+    return new JsonResponse($this->getBranches());
+  }
+
+  /**
+   * Provides a JSON response containing a list of published classes.
+   *
+   * This method executes a database query to select node IDs (nid) and titles
+   * from the 'node_field_data' table where nodes are of the 'class' content
+   * type and have a status indicating they are published.
+   *
+   * @return JsonResponse
+   *   A JsonResponse object that contains the classes in an associative array
+   *   format, where each class ID (nid) is a key, and its corresponding title
+   *   is the value.
+   */
+  public function getClassesOptions(): JsonResponse {
+    $query = $this->database->select('node_field_data', 'n');
+    $query->fields('n', ['nid', 'title']);
+    $query->condition('n.status', NodeInterface::PUBLISHED);
+    $query->condition('n.type', 'class');
+    $query->orderBy('n.title');
+    $res = $query->execute()->fetchAllKeyed();
+
+    return new JsonResponse($res);
+  }
+
+  /**
+   * Retrieves a list of published activity categories from the database and
+   * provides them in a JSON response.
+   *
+   * This method queries the 'node_field_data' table to select node IDs (nid)
+   * and titles for nodes of the 'activity' content type that are published.
+   *
+   * @return JsonResponse
+   *   A JsonResponse object containing an array of categories. Each category
+   *   is represented as an associative array with 'name' and 'color' keys,
+   *   where 'name' corresponds to the category title and 'color' is a
+   *   placeholder value.
+   */
+  public function getCategories(): JsonResponse {
+    $query = $this->database->select('node_field_data', 'n');
+    $query->fields('n', ['nid', 'title']);
+    $query->condition('n.status', NodeInterface::PUBLISHED);
+    $query->condition('n.type', 'activity');
+    $query->orderBy('n.title');
+    $result = $query->execute();
+
+    $categories = [];
+    foreach ($result as $record) {
+      // TODO: We assume that the color will be determined later, so for now we put a placeholder.
```

Comment:
Let's have a ticket in Jira
Also - to define this in config for better maintainability

Reply:
In the description to the ticket it is already said that we have to store color in the field and map it to the color on calendar.
This is important to do in order to pass QA.

Reply:
This color in this case is the color for checkboxes.
Color for events is already implemented and for this a new field has been added to the session. This is one of the points I wanted to discuss, since with the current implementation we can control the color of each event, but based on the designs it is not necessary, the color should be bound for the category, if so then the checkboxes and events will automatically have colors.

Reply:
This needs to be discussed first.